snorkel/learning/disc_learning.py

```diff
@@ -205,8 +205,7 @@ def train(self, X_train, Y_train, n_epochs=25, lr=0.01, batch_size=256,
                 epoch_losses.append(epoch_loss)
 
             # Reshuffle training data
-            train_idxs = range(n)
-            np.random.shuffle(train_idxs)
+            train_idxs = np.random.permutation(n)
```

Comment:
Not Python 3 related but does appear to do the same thing as before with more brevity.